src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -207,3 +209,22 @@ export const sumUsdBalances = ([usdBalance1, usdBalance2]: [
  */
 export const isRequiredTokenWithLinkedData = (token: Token): token is RequiredTokenWithLinkedData =>
 	'twinTokenSymbol' in token && typeof token.twinTokenSymbol === 'string';
+
+/** Find in the provided tokens list a twin IC token by symbol.
+ *
+ * @param tokenToPair - token to find a twin for.
+ * @param tokens - The list of tokens.
+ * @returns IcCkToken or undefined if no twin token found.
+ * */
+export const findTwinToken = ({
+	tokenToPair,
+	tokens
+}: {
+	tokenToPair: Token;
+	tokens: Token[];
+}): IcCkToken | undefined =>
+	isRequiredTokenWithLinkedData(tokenToPair)
+		? (tokens.find(
+				(token) => token.symbol === tokenToPair.twinTokenSymbol && isIcCkToken(token)
+			) as IcCkToken | undefined)
```

Comment:
I just noticed one thing, sorry for the late note:

can you use `IcCkTokenSchema.safeParse` here?

Reply:
I am now using `import { isIcCkToken } from '$icp/validation/ic-token.validation';` to validate if it's a IcCkToken. Is this something you had in mind?

Reply:
even better! thank you!